src/Pdf.php

```diff
@@ -12,38 +13,63 @@ class Pdf
 
     protected $binPath;
 
+    protected $options = [];
+
     public function __construct(string $binPath = null)
     {
         $this->binPath = $binPath ?? '/usr/bin/pdftotext';
     }
 
-    public function setPdf(string $pdf) : Pdf
+    public function setPdf(string $pdf) : self
     {
-        if (!file_exists($pdf)) {
-            throw new PdfNotFound("could not find pdf {$pdf}");
+        if (!is_readable($pdf)) {
+            throw new PdfNotFound(sprintf('could not find or read pdf `%s`', $pdf));
         }
 
         $this->pdf = $pdf;
 
         return $this;
     }
 
+    public function setOptions(array $options) : self
+    {
+        $this->options = array_map([$this, 'formatOption'], $options);
+
+        return $this;
+    }
+
+    protected function formatOption(string $content) : string
+    {
+        $content = trim($content);
+        if ('-' === $content[0] ?? '') {
+            return $content;
+        }
+
+        return '-'.$content;
+    }
+
     public function text() : string
     {
-        $process = new Process("{$this->binPath} " . escapeshellarg($this->pdf) . " -");
-        $process->run();
+        $arguments = $this->options;
+        $arguments[] = escapeshellarg($this->pdf);
+        $arguments[] = '-';
 
+        $commandline = $this->binPath.' '.implode(' ', $arguments);
```

Comment:
Would it be possible to use and array with the arguments here, instead of concatenating to a string?

See: https://symfony.com/doc/current/components/process.html

Reply:
@freekmurze that's not possible because of your travis settings. You minimum requirement for symfony/process do not allow the use of array arguments 😢 unless you update the project requirements which means BC break in my book.